lib/vdsm/virt/cpumanagement.py

```diff
@@ -160,6 +163,37 @@ def libvirt_cpuset_spec(cpus, cpu_list_length):
     return tuple(cpuset)
 
 
+def resume_cpu_pinning(from_engine, stored):
+    """
+    Update CPU pinning in domain XML of VM resumed from hibernation.
+
+    :param from_engine: domain XML passed from engine
+    :type from_engine: str
+    :param stored: domain XML stored in hibernation volume of VM
+    :param stored: str
+    """
+    if '_srcDomXML' not in stored or 'xml' not in from_engine:
+        return stored
+    dom_stored = xmlutils.fromstring(stored['_srcDomXML'])
+    dom_engine = xmlutils.fromstring(from_engine['xml'])
+    # Remove any previous pinning
+    cputune_stored = dom_stored.find('cputune')
+    if cputune_stored is not None:
+        for vcpu in vmxml.find_all(cputune_stored, 'vcpupin'):
+            cputune_stored.remove(vcpu)
+    else:
+        cputune_stored = ET.Element('cputune')
+        dom_stored.append(cputune_stored)
+    # Move pinning we have from engine
+    cputune_engine = dom_engine.find('cputune')
+    if cputune_engine is not None:
+        for vcpu in vmxml.find_all(cputune_engine, 'vcpupin'):
+            cputune_engine.remove(vcpu)
+            cputune_stored.append(vcpu)
+    stored['_srcDomXML'] = xmlutils.tostring(dom_stored)
+    return stored
```

Comment:
Since `stored` is modified by side-effect, it's just confusing to return it.  I'd suggest either not to return anything or to return the new _srcDomXML without modifying `stored`.